src/Helper.s

```diff
@@ -50,16 +84,22 @@ SDCMD:      PHY
 ;
 ;*******************************
 
-GETR1:      LDA   #DUMMY
-            STA   DATA,X
-@WAIT:      BIT   CTRL,X
-            BPL   @WAIT
+GETR1:      PHY
+            LDY   #10
+@AGAIN:     LDA   #DUMMY
+            JSR   WRDATA
             LDA   DATA,X      ; get response
-            BMI   GETR1       ; wait for MSB=0
-            PHA
+            BIT   #$80
```

Comment:
To test and branch on the highest bit, you may use BMI (MSB=1) and BPL (MSB=0). This saves an instruction

Reply:
Agreed,thanks